### PR TITLE
convert templates to YAML format

### PR DESCRIPTION
**Pull request by JaydenLiang**

### CODE REVIEW

**Review by jamie-pate — DISMISSED** [review]

**Review by Joel-Cripps — DISMISSED** [review]

File: templates/autoscale-existing-vpc.template.yaml [hunk 1]
```diff
@@ -0,0 +1,809 @@
...
+      types, see https://aws.amazon.com/ec2/instance-types/.
+  FortiOSVersion:
+    Type: String
+    Default: 6.2.3
+    AllowedValues:
+      - 6.2.3
+    ConstraintDescription: must be a valid FortiOS version from the selection.
+    Description: >-
+      The FortiOS version supported by Fortinet FortiGate Auto Scaling.
+      **IMPORTANT!** Requires a subscription to the Fortinet FortiGate On-demand
```
Comment (klc178) [inline templates/autoscale-existing-vpc.template.yaml]:
```suggestion
      **IMPORTANT!** Requires a subscription to the Fortinet FortiGate On-Demand
```

File: templates/autoscale-existing-vpc.template.yaml [hunk 2]
```diff
@@ -0,0 +1,809 @@
...
+  FgtAsgDesiredCapacityByol:
+    Type: Number
+    Default: 2
+    MinValue: 0
+    ConstraintDescription: must be a valid number not less than 0.
+    Description: >-
+      The number of FortiGate instances the BYOL Auto Scaling group should have
+      at any time. For High Availability in BYOL-only and Hybrid use cases,
+      ensure at least 2 FortiGates are in the group. For specific use cases, set
+      to 0 for On-demand-only, and >= 2 for BYOL-only or hybrid licensing.
```
Comment (klc178) [inline templates/autoscale-existing-vpc.template.yaml]:
```suggestion
      to 0 for On-Demand-only, and >= 2 for BYOL-only or hybrid licensing.
```

File: templates/autoscale-existing-vpc.template.yaml [hunk 3]
```diff
@@ -0,0 +1,809 @@
...
+      Minimum number of FortiGate instances in the On-Demand Auto Scaling group.
+      For specific use cases, set to 0 for BYOL-only, >= 2 for On-Demand-only,
+      and >= 0 for hybrid licensing.
+  FgtAsgMaxSizePayg:
+    Type: Number
+    Default: 6
+    MinValue: 0
+    ConstraintDescription: must be a valid number not less than 0.
+    Description: >-
+      The maximum number of FortiGate instances in the On-demand Auto Scaling
```
Comment (klc178) [inline templates/autoscale-existing-vpc.template.yaml]:
```suggestion
      The maximum number of FortiGate instances in the On-Demand Auto Scaling
```

File: templates/autoscale-existing-vpc.template.yaml [hunk 4]
```diff
@@ -0,0 +1,809 @@
...
+      group. For specific use cases, set to 0 for BYOL-only, >= 2 for
+      On-demand-only, and >= 0 for hybrid licensing. This number must be greater
+      than or equal to the Minimum group size (On-demand).
```
Comment (klc178) [inline templates/autoscale-existing-vpc.template.yaml]:
```suggestion
      than or equal to the Minimum group size (On-Demand).
```

File: templates/autoscale-existing-vpc.template.yaml [hunk 5]
```diff
@@ -0,0 +1,809 @@
...
+      - m5.xlarge
+      - m5.2xlarge
+      - m5.4xlarge
+      - m5.12xlarge
+      - t2.medium
+      - t2.large
+      - t2.xlarge
+    ConstraintDescription: must be a valid EC2 instance type.
+    Description: >-
+      The instance type to launch as FortiAnalyzer On-demand instances. There
```
Comment (klc178) [inline templates/autoscale-existing-vpc.template.yaml]:
```suggestion
      The instance type to launch as FortiAnalyzer On-Demand instances. There
```

File: templates/autoscale-existing-vpc.template.yaml [hunk 6]
```diff
@@ -0,0 +1,809 @@
...
+      FgtAsgDesiredCapacityByol:
+        default: Desired capacity (BYOL)
+      FgtAsgMinSizeByol:
+        default: Minimum group size (BYOL)
+      FgtAsgMaxSizeByol:
+        default: Maximum group size (BYOL)
+      FgtAsgDesiredCapacityPayg:
+        default: Desired capacity (On-demand)
+      FgtAsgMinSizePayg:
+        default: Minimum group size (On-demand)
```
Comment (klc178) [inline templates/autoscale-existing-vpc.template.yaml]:
```suggestion
        default: Minimum group size (On-Demand)
```

File: templates/autoscale-existing-vpc.template.yaml [hunk 7]
```diff
@@ -0,0 +1,809 @@
...
+      FgtAsgCooldown:
+        default: Scaling cooldown period
+      FgtAsgDesiredCapacityByol:
+        default: Desired capacity (BYOL)
+      FgtAsgMinSizeByol:
+        default: Minimum group size (BYOL)
+      FgtAsgMaxSizeByol:
+        default: Maximum group size (BYOL)
+      FgtAsgDesiredCapacityPayg:
+        default: Desired capacity (On-demand)
```
Comment (klc178) [inline templates/autoscale-existing-vpc.template.yaml]:
```suggestion
        default: Minimum group size (On-Demand)
```

File: templates/autoscale-existing-vpc.template.yaml [hunk 8]
```diff
@@ -0,0 +1,809 @@
...
+        - '-'
+        - !Select
+          - 2
+          - !Split
+            - /
+            - !Ref 'AWS::StackId'
+  FgtLicensingModel:
+    Description: >-
+      The FortiGate licensing model in the Auto Scaling group(s) for the initial
+      deployment of this stack. (Options: On-demand-only, BYOL-only, Hybrid)
```
Comment (klc178) [inline templates/autoscale-existing-vpc.template.yaml]:
```suggestion
      deployment of this stack. (Options: On-Demand-only, BYOL-only, Hybrid)
```

File: templates/autoscale-new-vpc.template.yaml [hunk 9]
```diff
@@ -0,0 +1,874 @@
...
+      types, see https://aws.amazon.com/ec2/instance-types/.
+  FortiOSVersion:
+    Type: String
+    Default: 6.2.3
+    AllowedValues:
+      - 6.2.3
+    ConstraintDescription: must be a valid FortiOS version from the selection.
+    Description: >-
+      The FortiOS version supported by Fortinet FortiGate Auto Scaling.
+      **IMPORTANT!** Requires a subscription to the Fortinet FortiGate On-demand
```
Comment (klc178) [inline templates/autoscale-new-vpc.template.yaml]:
```suggestion
      **IMPORTANT!** Requires a subscription to the Fortinet FortiGate On-Demand
```

File: templates/autoscale-new-vpc.template.yaml [hunk 10]
```diff
@@ -0,0 +1,874 @@
...
+  FgtAsgDesiredCapacityByol:
+    Type: Number
+    Default: 2
+    MinValue: 0
+    ConstraintDescription: must be a valid number not less than 0.
+    Description: >-
+      The number of FortiGate instances the BYOL Auto Scaling group should have
+      at any time. For High Availability in BYOL-only and Hybrid use cases,
+      ensure at least 2 FortiGates are in the group. For specific use cases, set
+      to 0 for On-demand-only, and >= 2 for BYOL-only or hybrid licensing.
```
Comment (klc178) [inline templates/autoscale-new-vpc.template.yaml]:
```suggestion
      to 0 for On-Demand-only, and >= 2 for BYOL-only or hybrid licensing.
```

File: templates/autoscale-new-vpc.template.yaml [hunk 11]
```diff
@@ -0,0 +1,874 @@
...
+  FgtAsgMinSizeByol:
+    Type: Number
+    Default: 2
+    MinValue: 0
+    ConstraintDescription: must be a valid number not less than 0.
+    Description: >-
+      The minimum number of FortiGate instances in the BYOL Auto Scaling group.
+      For specific use cases, set to 0 for On-demand-only, and >= 2 for
```
Comment (klc178) [inline templates/autoscale-new-vpc.template.yaml]:
```suggestion
      For specific use cases, set to 0 for On-Demand-only, and >= 2 for
```

File: templates/autoscale-new-vpc.template.yaml [hunk 12]
```diff
@@ -0,0 +1,874 @@
...
+      BYOL-only or hybrid licensing.
+  FgtAsgMaxSizeByol:
+    Type: Number
+    Default: 2
+    MinValue: 0
+    ConstraintDescription: must be a valid number not less than 0.
+    Description: >-
+      The maximum number of FortiGate instances in the BYOL Auto Scaling group.
+      For specific use cases, set to 0 for On-demand-only, and >= 2 for
```
Comment (klc178) [inline templates/autoscale-new-vpc.template.yaml]:
```suggestion
      For specific use cases, set to 0 for On-Demand-only, and >= 2 for
```

File: templates/autoscale-new-vpc.template.yaml [hunk 13]
```diff
@@ -0,0 +1,874 @@
...
+      BYOL-only or hybrid licensing. This number must be greater than or equal
+      to the Minimum group size (BYOL).
+  FgtAsgDesiredCapacityPayg:
+    Type: Number
+    Default: 0
+    MinValue: 0
+    ConstraintDescription: must be a valid number not less than 0.
+    Description: >-
+      The number of FortiGate instances the On-demand Auto Scaling group should
```
Comment (klc178) [inline templates/autoscale-new-vpc.template.yaml]:
```suggestion
      The number of FortiGate instances the On-Demand Auto Scaling group should
```

File: templates/autoscale-new-vpc.template.yaml [hunk 14]
```diff
@@ -0,0 +1,874 @@
...
+      have at any time. For High Availability in an On-demand-only use case,
```
Comment (klc178) [inline templates/autoscale-new-vpc.template.yaml]:
```suggestion
      have at any time. For High Availability in an On-Demand-only use case,
```

File: templates/autoscale-new-vpc.template.yaml [hunk 15]
```diff
@@ -0,0 +1,874 @@
...
+      ensure at least 2 FortiGates are in the group. For specific use cases, set
+      to 0 for BYOL-only, >= 2 for On-demand-only, and >= 0 for hybrid
```
Comment (klc178) [inline templates/autoscale-new-vpc.template.yaml]:
```suggestion
      to 0 for BYOL-only, >= 2 for On-Demand-only, and >= 0 for hybrid
```

File: templates/autoscale-new-vpc.template.yaml [hunk 16]
```diff
@@ -0,0 +1,874 @@
...
+      licensing.
+  FgtAsgMinSizePayg:
+    Type: Number
+    Default: 0
+    MinValue: 0
+    ConstraintDescription: must be a valid number not less than 0.
+    Description: >-
+      The minimum number of FortiGate instances in the On-demand Auto Scaling
```
Comment (klc178) [inline templates/autoscale-new-vpc.template.yaml]:
```suggestion
      The minimum number of FortiGate instances in the On-Demand Auto Scaling
```

File: templates/autoscale-new-vpc.template.yaml [hunk 17]
```diff
@@ -0,0 +1,874 @@
...
+      group. For specific use cases, set to 0 for BYOL-only, >= 2 for
+      On-demand-only, and >= 0 for hybrid licensing.
```
Comment (klc178) [inline templates/autoscale-new-vpc.template.yaml]:
```suggestion
      On-Demand-only, and >= 0 for hybrid licensing.
```

File: templates/autoscale-new-vpc.template.yaml [hunk 18]
```diff
@@ -0,0 +1,874 @@
...
+  FgtAsgMaxSizePayg:
+    Type: Number
+    Default: 6
+    MinValue: 0
+    ConstraintDescription: must be a valid number not less than 0.
+    Description: >-
+      The maximum number of FortiGate instances in the On-demand Auto Scaling
```
Comment (klc178) [inline templates/autoscale-new-vpc.template.yaml]:
```suggestion
      The maximum number of FortiGate instances in the On-Demand Auto Scaling
```

File: templates/autoscale-new-vpc.template.yaml [hunk 19]
```diff
@@ -0,0 +1,874 @@
...
+      group. For specific use cases, set to 0 for BYOL-only, >= 2 for
+      On-demand-only, and >= 0 for hybrid licensing. This number must be greater
```
Comment (klc178) [inline templates/autoscale-new-vpc.template.yaml]:
```suggestion
      On-Demand-only, and >= 0 for hybrid licensing. This number must be greater
```

File: templates/autoscale-new-vpc.template.yaml [hunk 20]
```diff
@@ -0,0 +1,874 @@
...
+      FgtAsgCooldown:
+        default: Scaling cooldown period
+      FgtAsgDesiredCapacityByol:
+        default: Desired capacity (BYOL)
+      FgtAsgMinSizeByol:
+        default: Minimum group size (BYOL)
+      FgtAsgMaxSizeByol:
+        default: Maximum group size (BYOL)
+      FgtAsgDesiredCapacityPayg:
+        default: Desired capacity (On-demand)
```
Comment (klc178) [inline templates/autoscale-new-vpc.template.yaml]:
```suggestion
        default: Desired capacity (On-Demand)
```

File: templates/autoscale-new-vpc.template.yaml [hunk 21]
```diff
@@ -0,0 +1,874 @@
...
+      FgtAsgMinSizePayg:
+        default: Minimum group size (On-Demand)
+      FgtAsgMaxSizePayg:
+        default: Maximum group size (On-demand)
```
Comment (klc178) [inline templates/autoscale-new-vpc.template.yaml]:
```suggestion
        default: Maximum group size (On-Demand)
```

**Review by klc178 — COMMENTED** [review]
AWS uses "On-Demand". I've hopefully highlighted all instances of "On-demand"
What is the work effort involved to change the ....Payg variables to ...OnDemand?

**Review by jamie-pate — APPROVED** [review]

**Review by klc178 — APPROVED** [review]

**Pull request merged**